### PR TITLE
Sighting by location

### LINKED ISSUE #37: BE Sightings Query Refactor
refactor sightings query

### PR DESCRIPTION
**Pull request by ParkerLockhart**
## Neccesary checkmarks:

    [x] All Tests are Passing in all environments

    [x] The code will run locally

## Type of change

    [x] New feature
    [] Bug Fix
    [] Refactor

## Description of Change:

    description: 
- Sightings refactor
- sightings by location query

## Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

## Testing Changes

    [x] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

    [x] My code has no unused/commented out code
    [x] My code has no binding.pry's
    [x] I have reviewed my code
    [x] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

closes #37  
closes #52

### CODE REVIEW

**Review by dkulback — APPROVED** [review]

File: app/graphql/queries/sightings.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Queries
+  class Sightings < Queries::BaseQuery
+    description 'All sightings'
+
+    type [Types::SightingType], null: false
+
+    def resolve
+      ::Sighting.all
```
Comment (BrianZanti) [inline app/graphql/queries/sightings.rb:8]: I'm wondering what the difference here would be between `::Sighting.all` and `Sighting.all`?

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -9,19 +9,8 @@ class QueryType < Types::BaseObject
...
-    field :sighting_by_id, SightingType, "get Sighting by ID" do
-      argument :id, ID
-    end
-
-    def sighting_by_id(args)
-      Sighting.find(args[:id])
-    end
+    field :sightings, resolver: Queries::Sightings
+    field :sighting_by_id, resolver: Queries::SightingById
+    field :sighting_by_location, resolver: Queries::SightingByLocation
```
Comment (BrianZanti) [inline app/graphql/types/query_type.rb:14]: I like this change. This class feels much more single responsibility.

File: app/models/sighting.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
   belongs_to :cryptid
   validates :title, :description, :location, presence: true
+
+  def self.location_search(search_term)
+    if search_term.blank?
+      []
+    else
+      where('location ILIKE ?', "%#{search_term}%")
+    end
+  end
```
Comment (BrianZanti) [inline app/models/sighting.rb:11]: Looks good! Nice job handling the edge case.

File: spec/graphql/queries/sightings/sighting_by_location_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+module Queries
+  RSpec.describe SightingByLocation, type: :request do
```
Comment (BrianZanti) [inline spec/graphql/queries/sightings/sighting_by_location_spec.rb:4]: This tests look good, but I'm wondering if it would also be good to have "Request" level tests. For example, a test where you send a `post` request to `/graphql` with the required graphql query string in the body. That would more clearly illustrate what would be expected from the Frontend.

File: spec/models/sighting_spec.rb [hunk 5]
```diff
@@ -10,4 +10,17 @@
     it { should validate_presence_of(:title) }
     it { should validate_presence_of(:location) }
   end
+
+  describe 'class methods' do
+    describe 'location_search' do
+      it 'returns array of sightings by location' do
```
Comment (BrianZanti) [inline spec/models/sighting_spec.rb:16]: This test looks good. I think you should also add tests for the edge case when there is an empty query string and also test the case insensitive functionality you added to the method.

File: app/graphql/queries/sighting_by_id.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+module Queries
+  class SightingById < Queries::BaseQuery
```
Comment (BrianZanti) [inline app/graphql/queries/sighting_by_id.rb:2]: I like how you are separating out all the queries into their own files.

**Review by BrianZanti — CHANGES_REQUESTED** [review]
Hey everyone, I added some comments with requests for additional testing. I think the model test should be pretty straight forward. I also think the "request" level test will be especially valuable since it will illustrate what is expected from the FE. Overall though, I think you are doing some great work! And I like that you are refactoring the code to be more SRP rather than having one giant file for all the queries. I think that will really help in the long run.

**Review by Malllll12 — APPROVED** [review]
